imagebuildah/stage_executor.go

```diff
@@ -297,7 +297,17 @@ func (s *StageExecutor) digestSpecifiedContent(node *parser.Node) (string, error
 			}
 		}
 	}
+
 	for _, src := range srcs {
+		// If a src is an environment variable, try to resolve it.
+		if strings.HasPrefix(src, "$") {
```

Comment:
Does this need to be conditional?

Reply:
It doesn't need to be, meaning the call to ProcessWord will work, but I thought it would be better if we did the check.  Otherwise, we'll hit up ProcessWord for a Dockerfile line like `COPY myfile.txt /tmp/myfile.txt` and thought I'd skip the extra processing.  But if you think it might prove safer to remove the if statement here, I will.

Reply:
Mainly I want to avoid getting tripped up by people using arguments in the middle of values, or at the end for file extensions or things I haven't thought of.

Reply:
hmm, I'd not considered the old argument in the middle of a variable trick.  Sold, I'll remove the check for the beginning '$' and will whack them all.  Thanks!